actions/user_actions.jsx

```diff
@@ -138,9 +138,9 @@ export function loadChannelMembersForProfilesList(profiles, channelId) {
     };
 }
 
-export async function loadNewDMIfNeeded(channelId) {
+export async function loadNewDMIfNeeded(channelId, currentUserId) {
```

Comment:
It looks like you still have some work to do on this function since it still uses `getState` further down, but if you turn this into a redux action creator, you won't need to pass in the current user ID since you can get it from inside the function.
```javascript
export function loadNewDMIfNeeded(channelId) {
    return async (dispatch, getState) => {
        const currentUserId = getCurrentUserId();
        // ...
    };
}
```

If you do that though, you'll need to update everywhere where these functions are used to either pass it into the component using `mapDispatchToProps` or by dispatching it like `dispatch(laodNewDMIfNeeded(channelId));`